docs/ai/ai-extensions.md

```diff
@@ -0,0 +1,100 @@
+---
+title:  Unified AI building blocks for .NET
```

Comment:
This is fine as a separate document for now, but would it make sense for this to eventually be part of the overview as we pivot to focus on M.E.AI in this docset?

Reply:
I plan to add smaller content updates to other parts of these docs such as the ecosystem/tooling docs to wrap in MEAI more cohesively, that'll be a separate PR. Eventually yes this should be a bigger focus though.